### DISCUSSION TITLE
收藏功能体验优化

### DISCUSSION THREAD

**Discussion by lyppiao** [ywmoyue/biliuwp-lite#683] [feature]
### 问题描述

在拥有很多个收藏夹的情况下，收藏视频不能直接点击保存键，要拉到最底部才能看到保存键。
另外，保存后 收藏夹选择界面 不会自动收回，要额外点击其他无关区域才会消失。

### 描述解决方案

1.增加保存功能的快捷键。最好有两个，键盘快捷键 以及 鼠标中/右键。这样就比较灵活了，笔记本和台机都比较方便。
目前鼠标中、右键的逻辑是：“选择收藏夹“界面内 点击无反应，其他区域点击后退出选择界面。
那正好可以把这个改成 区域内“点击后保存“，区域外退出选择界面（维持不变）
2.保存后自动退出收藏夹选择界面。

### 代码贡献

否

**Comment by lyppiao** [comment]
另外我发现拖到最底部保存后，下次收藏视频的时候，选择界面就默认拉到最底端了。
这个没有必要，还是默认顶端的好。
因为一般常用收藏夹都是放在上面的，不常用的才在底下（这个在pc网页端可以自行排序的）。

**Reply by ywmoyue** [reply]
在我的收藏页面中也可以给收藏夹排序

现在问题其实应该是这边没给收藏夹列表一个最大高度，给一个最大高度后就是收藏夹列表本身滚动，不影响上下的内容比如保存按钮

**Reply by ywmoyue** [reply]
试试 https://github.com/ywmoyue/biliuwp-lite/releases/tag/v4.6.33.3401

**Reply by lyppiao** [reply]
> 试试https://github.com/ywmoyue/biliuwp-lite/releases/tag/v4.6.33.3401

感觉双键没有必要，直接用 S键 就行了。
当前视频把收藏夹拉到底部后，下一个视频收藏的时候还是默认弹出底部，很不方便。建议改成每个视频都重置到顶部。

用鼠标中键或右键 来保存，这个方便做吗?

目前主要是美观性问题，有两个方向可以参考下
1.把拖条往右边移动，不要挡住文字；
2.把文字往左边挪一点，跟保存键用居中对齐（如果保存键小了，可以加长一点）。
另外顶部的“创建”按键，也可以一起对齐掉。会更美观。
保存 单个按键，或者 保存+创建 两个键，有没有必要加粗？这个我不好判断。
![屏幕截图 2024-06-23 144901](https://github.com/ywmoyue/biliuwp-lite/assets/143381863/9d3316fe-f1f3-4e7a-b00f-8be0d6bb39cb)
![屏幕截图 2024-06-23 144931](https://github.com/ywmoyue/biliuwp-lite/assets/143381863/83b70bc2-3d59-495a-a437-6f8b306441fa)

**Reply by ywmoyue** [reply]
> > 试试https://github.com/ywmoyue/biliuwp-lite/releases/tag/v4.6.33.3401
> 
> 感觉双键没有必要，直接用 S键 就行了。 当前视频把收藏夹拉到底部后，下一个视频收藏的时候还是默认弹出底部，很不方便。建议改成每个视频都重置到顶部。


现在新版本可以自定义保存快捷键，自己改成S键就行了

**Comment by ywmoyue** [comment]
https://github.com/ywmoyue/biliuwp-lite/releases/tag/v4.6.33.3402 试试

**Reply by lyppiao** [reply]
> https://github.com/ywmoyue/biliuwp-lite/releases/tag/v4.6.33.3402 试试

挺好。
就是在稍后再看、视频合集里面，播放其他视频的时候，不会把收藏夹重置到顶部。
感觉这个逻辑是同一窗口不重置。像历史记录、视频推荐这些另开窗口的，就很正常。

另外我发现稍后再看有个bug，有时候播完视频以后，不会自动跳转下一个视频，有时候又能正常跳，很奇怪。
